qt/scientific_interfaces/Inelastic/Common/OutputPlotOptionsModel.cpp

```diff
@@ -237,26 +239,37 @@ void OutputPlotOptionsModel::plotTiled() {
   auto const workspaceName = workspace();
   auto const indicesString = indices();
   if (workspaceName && indicesString)
-    m_plotter->plotTiled(workspaceName.get(), indicesString.get(), SettingsHelper::externalPlotErrorBars());
+    m_plotter->plotTiled(workspaceName.value(), indicesString.value(), SettingsHelper::externalPlotErrorBars());
 }
 
-boost::optional<std::string> OutputPlotOptionsModel::singleDataPoint(MantidAxis const &axisType) const {
+std::optional<std::string> OutputPlotOptionsModel::singleDataPoint(MantidAxis const &axisType) const {
   if (auto const workspaceName = workspace())
-    return checkWorkspaceSize(workspaceName.get(), axisType);
-  return boost::none;
+    return checkWorkspaceSize(workspaceName.value(), axisType);
+  return std::nullopt;
 }
 
-boost::optional<std::string> OutputPlotOptionsModel::checkWorkspaceSize(std::string const &workspaceName,
-                                                                        MantidAxis const &axisType) const {
+std::optional<std::string> OutputPlotOptionsModel::checkWorkspaceSize(std::string const &workspaceName,
+                                                                      MantidAxis const &axisType) const {
   auto &ads = AnalysisDataService::Instance();
   if (ads.doesExist(workspaceName)) {
     if (auto const matrixWs = ads.retrieveWS<MatrixWorkspace>(workspaceName)) {
-      if (axisType == MantidAxis::Spectrum)
-        return checkWorkspaceSpectrumSize(matrixWs);
-      return checkWorkspaceBinSize(matrixWs);
+      if (axisType == MantidAxis::Spectrum) {
+        auto msg = checkWorkspaceBinSize(matrixWs);
+        return msg.has_value() ? "Plot Spectra Failed: " + msg.value() : msg;
+      } else if (axisType == MantidAxis::Bin) {
```

Comment:
Thanks for the changes, it all looks really good! Just spotted what I think is a small mistake then I'm happy to approve
```suggestion
      if (axisType == MantidAxis::Spectrum) {
        auto msg = checkWorkspaceSpectrumSize(matrixWs);
        return msg.has_value() ? "Plot Spectra Failed: " + msg.value() : msg;
      } else if (axisType == MantidAxis::Bin) {
```

Reply:
you are right!